src/AppBundle/Entity/Repository/ClientRepository.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace AppBundle\Entity\Repository;
+
+use AppBundle\Entity\Client;
+use Doctrine\ORM\EntityRepository;
+use AppBundle\Entity\User;
+
+/**
+ * ClientRepository.
+ *
+ * This class was generated by the Doctrine ORM. Add your own custom
+ * repository methods below.
+ */
+class ClientRepository extends EntityRepository
+{
+
+    /**
+     * Search Clients
+     *
+     * @param string $query Search query
+     * @param string $orderBy field to order by
+     * @param string $sortOrder order of field order ASC|DESC
+     * @param int $limit number of results to return
+     * @param string $offset
+     *
+     * @return Client[]|array
+     */
+    public function searchClients($query = '', $orderBy = 'lastname', $sortOrder = 'ASC', $limit = 100, $offset = 'id')
+    {
+
+        $qb = $this->createQueryBuilder('c');
+        $qb->setFirstResult($offset);
+        $qb->setMaxResults($limit);
+        $qb->leftJoin('c.users', 'u')
+            ->where('u.roleName = \'' . User::ROLE_LAY_DEPUTY . '\'');
+        $qb->orderBy('c.' . $orderBy, $sortOrder);
+
+        if ($query) {
+            if (preg_match('/^[0-9t]{8}$/i', $query)) { // case number
```

Comment:
minor: this is useful and could be moved to a static method into the Client class, and re-used. Might be used already around

Reply:
it IS used for the user search, but one issue we've got is that none of our test data matches it lol. as they all have case numbers with the word "test" so no room for numbers

Reply:
Moved to Client::isValidCaseNumber($q)